packages/node/src/lib/process-request.ts

```diff
@@ -207,14 +210,16 @@ export default function processRequest(
     queryString: searchToArray(reqUrl.searchParams),
     postData,
     // TODO: When readme starts accepting these, send the correct values
-    cookies: [],
+    cookies: [] as Cookie[],
     headersSize: -1,
     bodySize: -1,
   };
 
   if (requestData.postData === null) {
-    delete requestData.postData;
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { postData: postDataToBeOmitted, ...remainingRequestData } = requestData;
+    return remainingRequestData as Request;
   }
 
-  return requestData;
+  return requestData as Request;
```

Comment:
You could remove this cast if we can pass `castToString` for `headers` and make `method` a string:

```
   const requestData = {
-    method: req.method,
+    method: req.method || '',
     url: url.format({
       // Handle cases where some reverse proxies put two protocols into x-forwarded-proto
       // This line does the following: "https,http" -> "https"
@@ -206,7 +206,7 @@ export default function processRequest(
       query: qs.parse(reqUrl.search.substring(1)),
     }),
     httpVersion: `${getProto(req).toUpperCase()}/${req.httpVersion}`,
-    headers: objectToArray(req.headers),
+    headers: objectToArray(req.headers, { castToString: true }),
     queryString: searchToArray(reqUrl.searchParams),
     postData,
     // TODO: When readme starts accepting these, send the correct values
@@ -221,5 +221,5 @@ export default function processRequest(
     return remainingRequestData as Request;
   }
 
-  return requestData as Request;
+  return requestData;
 }
```

I'm not 100% sure it's valid to convert the headers into strings? ISTM that it probably is but I don't know - I just dislike using `as Request` when possible

Reply:
yeah I like this suggestion but I'm wondering if there was a deliberate reason why we're not setting that `castToString` flag here. again this is another example of being unsure of what our ingestion service likes

Reply:
I'd be pretty concerned if we passed `{ method: '' }`

Reply:
Riiiight, but for some reason `req.method` can be undefined per the official Node types 🤔 I'll leave that fallback in there for now knowing that it probably won't get hit at all, but let me know if you disagree here!

Reply:
ok on second thought I decided to enable the `castToString` flag on this line:

https://github.com/readmeio/metrics-sdks/blob/1585f05c5877bf90e6d5f9c920cb160a2aaf3629/packages/node/src/lib/process-request.ts#L209

is this a safe change to make @gratcliff?